extensions/sql-database-projects/src/dialogs/addSqlBindingQuickpick.ts

```diff
@@ -8,29 +10,23 @@ export async function launchAddSqlBindingQuickpick(uri: vscode.Uri | undefined):
 		uri = vscode.window.activeTextEditor!.document.uri;
 	}
 
-	// 1. select input or output binding
-	const inputOutputItems: string[] = [constants.input, constants.output];
-
-	const selectedBinding = (await vscode.window.showQuickPick(inputOutputItems, {
-		canPickMany: false,
-		title: constants.selectBindingType,
-		ignoreFocusOut: true
-	}));
+	// get all the Azure functions in the file
+	const azureFunctionsService = await utils.getAzureFunctionService();
+	const getAzureFunctionsResult = await azureFunctionsService.getAzureFunctions(uri.fsPath);
```

Comment:
Was this the one that was going to throw if an error occurs? Regardless some other error could happen so consider wrapping this in a try/catch and displaying the error for that as well if it happens. (will you still need the success check at that point then? Do you ever actually return false?)

Reply:
Also how long does this call take? You may want to look into using a progress notification to indicate that it's loading so the user doesn't think it's hung while it's doing this (I do something similar when loading the profile in the publish quickpick 

https://github.com/microsoft/azuredatastudio/blob/main/extensions/sql-database-projects/src/dialogs/publishDatabaseQuickpick.ts#L22

It's not the most succinct code unfortunately but we could look into making a generic function for doing something like this since there's a few instances of loading something in the background (DBs for example)

Reply:
ah thanks I forgot to update this after the change in SqlToolsService to return an error instead of a result with success = false. 

It was instantaneous when I tried this on a file with 50 Azure functions, so I'll keep it as is for now.